design/authenticators/authn_k8s/inject_client_cert_solution_design.md

```diff
@@ -0,0 +1,255 @@
+# Solution Design - Inject Client Cert fix
+[//]: # "Change the title above from 'Template' to your design's title"
+
+## Table of Contents
+- [Useful links](#useful-links)
+- [Background](#background)
+- [Issue description](#issue-description)
+- [Solution](#solution)
+  * [Waiting for the injection to finish](#waiting-for-the-injection-to-finish)
+  * [Inject_client_cert response code](#inject-client-cert-response-code)
+  * [Cert injection implementation](#cert-injection-implementation)
+    + [Option 1 - Run a parallel request to verify cert injection](#option-1---run-a-parallel-request-to-verify-cert-injection)
+    + [Option 2 - Send the cert content as part of the URL](#option-2---send-the-cert-content-as-part-of-the-url)
+    + [Option 3 - Client sets its State once cert is injected](#option-3---client-sets-its-state-once-cert-is-injected)
+    + [Option 4 - Send cert as part of bash script](#option-4---send-cert-as-part-of-bash-script)
+    + [Conclusion](#conclusion)
+  * [Design](#design)
+  * [Backwards compatibility](#backwards-compatibility)
+  * [Performance](#performance)
+  * [Security](#security)
+- [Affected Components](#affected-components)
+- [Cross-team dependencies](#cross-team-dependencies)
+- [Test Plan](#test-plan)
+- [Logs](#logs)
+  * [Audit](#audit)
+- [Documentation](#documentation)
+- [Delivery plan](#delivery-plan)
+  * [Total](#total)
+  * [Parallelism](#parallelism)
+  * [Spikes and researches](#spikes-and-researches)
+  * [Implementation](#implementation)
+  * [Testing](#testing)
+  * [Demo](#demo)
+    
+## Useful links
+[//]: # "Add links that may be useful for the reader"
+
+- [Known issue in k8s exec](https://github.com/kubernetes/kubernetes/issues/89899)
+
+## Background
+[//]: # "Give relevant background for the designed feature. What is the motivation for this solution?"
```

Comment:
From meeting:
@oburstein-hub wanted clarification on error types. Clarify server-side and client-side errors for communication conjur server -> client
1. conjur server -> k8s api
2. k8s api -> authn client